### PR TITLE
[new mongodb SDK and standby instance]

### PR DESCRIPTION
**Pull request by brickzzhang**
1. supports standby instance
2. supports prepaid type for both replset and sharding instances
3. new mongodb SDK adaption for all of the datasource, replset and sharding instances

--------

# 副本集测试：

➜  terraform-provider-tencentcloud git:(master) ✗ make testacc TESTARGS="-run=TestAccTencentCloudMongodbInstanceResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudMongodbInstanceResource -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudMongodbInstanceResource
    TestAccTencentCloudMongodbInstanceResource: testing.go:745: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: errors during apply: PREPAID instances are not allowed to be deleted now, please isolate them on console

        State: tencentcloud_mongodb_instance.mongodb_prepaid:
          ID = cmgo-p46aysp9
          provider = provider.tencentcloud
          auto_renew_flag = 1
          available_zone = ap-guangzhou-3
          charge_type = PREPAID
          create_time = 2020-07-23 13:38:19
          engine_version = MONGO_40_WT
          instance_name = tf-mongodb-test-prepaid-update
          machine_type = HIO10G
          memory = 4
          password = test1234update
          prepaid_period = 1
          project_id = 0
          standby_instance_list.# = 0
          status = 2
          subnet_id =
          tags.% = 1
          tags.prepaid = prepaid
          vip = 10.66.192.187;10.66.237.191;10.66.89.99
          volume = 100
          vpc_id =
          vport = 27017
--- FAIL: TestAccTencentCloudMongodbInstanceResource (1204.40s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	1206.506s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]
FAIL
make: *** [testacc] Error 1

# 分片集测试：

➜  terraform-provider-tencentcloud git:(master) ✗ make testacc TESTARGS="-run=TestAccTencentCloudMongodbShardingInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudMongodbShardingInstance -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudMongodbShardingInstance
    TestAccTencentCloudMongodbShardingInstance: testing.go:745: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: errors during apply: PREPAID instances are not allowed to be deleted now, please isolate them on console

        State: tencentcloud_mongodb_sharding_instance.mongodb_prepaid:
          ID = cmgo-9olnycyp
          provider = provider.tencentcloud
          auto_renew_flag = 0
          available_zone = ap-guangzhou-3
          charge_type = PREPAID
          create_time = 2020-07-23 14:05:06
          engine_version = MONGO_40_WT
          instance_name = tf-mongodb-sharding-prepaid-update
          machine_type = HIO10G
          memory = 8
          nodes_per_shard = 3
          password = test1234update
          prepaid_period = 1
          project_id = 0
          shard_quantity = 2
          status = 2
          subnet_id =
          tags.% = 1
          tags.prepaid = prepaid
          vip = 10.66.184.131
          volume = 200
          vpc_id =
          vport = 27017
--- FAIL: TestAccTencentCloudMongodbShardingInstance (1053.44s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	1056.273s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]
FAIL
make: *** [testacc] Error 1

# 灾备实例测试：

➜  terraform-provider-tencentcloud git:(master) ✗ make testacc TESTARGS="-run=TestAccTencentCloudMongodbStandbyInstanceResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudMongodbStandbyInstanceResource -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	2.661s [no tests to run]
=== RUN   TestAccTencentCloudMongodbStandbyInstanceResource
    TestAccTencentCloudMongodbStandbyInstanceResource: testing.go:745: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: errors during apply: PREPAID instances are not allowed to be deleted now, please isolate them on console

        State: tencentcloud_mongodb_instance.mongodb_prepaid:
          ID = cmgo-09n9gzb7
          provider = provider.tencentcloud
          auto_renew_flag = 1
          available_zone = ap-guangzhou-3
          charge_type = PREPAID
          create_time = 2020-07-23 14:32:41
          engine_version = MONGO_40_WT
          instance_name = tf-mongodb-test-prepaid
          machine_type = HIO10G
          memory = 8
          password = test1234
          prepaid_period = 1
          project_id = 0
          standby_instance_list.# = 1
          standby_instance_list.0.standby_instance_id = cmgo-ni7syvd3
          standby_instance_list.0.standby_instance_region = ap-guangzhou
          status = 2
          subnet_id =
          tags.% = 1
          tags.test = test-prepaid
          vip = 10.66.191.228;10.66.191.84;10.66.219.147
          volume = 200
          vpc_id =
          vport = 27017
        tencentcloud_mongodb_standby_instance.mongodb_prepaid:
          ID = cmgo-ni7syvd3
          provider = provider.tencentcloud
          auto_renew_flag = 1
          available_zone = ap-guangzhou-4
          charge_type = PREPAID
          create_time = 2020-07-23 14:37:27
          engine_version = MONGO_40_WT
          father_instance_id = cmgo-09n9gzb7
          father_instance_region = ap-guangzhou
          instance_name = tf-mongodb-standby-prepaid-update
          machine_type = HIO10G
          memory = 4
          prepaid_period = 1
          project_id = 0
          status = 2
          subnet_id =
          tags.% = 1
          tags.prepaid = prepaid
          vip = 10.66.176.197;10.66.244.178;10.66.180.47
          volume = 100
          vpc_id =
          vport = 27017
--- FAIL: TestAccTencentCloudMongodbStandbyInstanceResource (2005.24s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	2007.838s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]
FAIL
make: *** [testacc] Error 1

# datasouce测试

➜  terraform-provider-tencentcloud git:(master) ✗ make testacc TESTARGS="-run=TestAccTencentCloudMongodbInstancesDataSource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudMongodbInstancesDataSource -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudMongodbInstancesDataSource
--- PASS: TestAccTencentCloudMongodbInstancesDataSource (412.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	412.611s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

### CODE REVIEW

File: tencentcloud/extension_mongodb.go [hunk 1]
```diff
@@ -16,20 +23,178 @@ const (
...
+		},
+		// payment
+		"charge_type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			Default:      MONGODB_CHARGE_TYPE_POSTPAID,
+			ValidateFunc: validateAllowedStringValue([]string{MONGODB_CHARGE_TYPE_POSTPAID, MONGODB_CHARGE_TYPE_PREPAID}),
+			Description:  "The charge type of instance. Valid values are `PREPAID` and `POSTPAID`. Default value is `POSTPAID`. Note: TencentCloud International only supports `POSTPAID`. Caution that update operation on this field will delete old instances and create new with new charge type.",
+		},
```
Comment (gailwang) [inline tencentcloud/extension_mongodb.go:166]: 数据库的charge_type 对齐 "PREPAID"  "POSTPAID_BY_HOUR"

File: tencentcloud/internal/helper/transform.go [hunk 2]
```diff
@@ -109,3 +111,8 @@ func BoolToInt64Ptr(s bool) (i *int64) {
 	i = &result
 	return
 }
+
+func CopySelf(client *connectivity.TencentCloudClient) *connectivity.TencentCloudClient {
+	tmpClient := *client
+	return &tmpClient
+}
```
Comment (Himer) [inline tencentcloud/internal/helper/transform.go:118]: ???

File: tencentcloud/resource_tc_mongodb_standby_instance_test.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  available_zone         = "ap-guangzhou-4"
+  project_id             = 0
+  charge_type            = "PREPAID"
+  prepaid_period         = 1
+  auto_renew_flag        = 1
+  father_instance_id     = tencentcloud_mongodb_instance.mongodb_prepaid.id
+  father_instance_region = "ap-guangzhou"
+
+  tags = {
+    "test" = "test-prepaid"
```
Comment (gailwang) [inline tencentcloud/resource_tc_mongodb_standby_instance_test.go:181]: test = “test-prepaid”

File: tencentcloud/extension_mongodb.go [hunk 4]
```diff
@@ -16,20 +23,178 @@ const (
...
+			Optional:    true,
+			ForceNew:    true,
+			Default:     "",
+			Description: "ID of the VPC.",
+		},
+		"subnet_id": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			ForceNew:    true,
+			Description: "ID of the subnet within this VPC. The vaule is required if VpcId is set.",
```
Comment (Himer) [inline tencentcloud/extension_mongodb.go:127]: The vaule is required if `vpc_id` is set.

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 5]
```diff
@@ -51,172 +78,165 @@ func resourceTencentCloudMongodbInstance() *schema.Resource {
...
+			securityGroups := v.(*schema.Set).List()
+			requestByMonth.SecurityGroup = make([]*string, 0, len(securityGroups))
+			for _, v := range securityGroups {
+				requestByMonth.SecurityGroup = append(requestByMonth.SecurityGroup, helper.String(v.(string)))
+			}
+		}
+	} else {
+		requestByUse.ReplicateSetNum = helper.IntUint64(replicateSetNum)
+		requestByUse.NodeNum = helper.IntUint64(nodeNum)
+		requestByUse.GoodsNum = helper.IntUint64(goodsNum)
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_instance.go:152]: requestByUse 和 requestByMonth只有两个属性不一样?  可以反射来赋值减少代码

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 6]
```diff
@@ -51,172 +78,165 @@ func resourceTencentCloudMongodbInstance() *schema.Resource {
...
+		}
+		if v, ok := d.GetOk("subnet_id"); ok {
+			requestByUse.SubnetId = helper.String(v.(string))
+		}
+		err := fmt.Errorf("you have to set vpc_id and subnet_id both")
+		if (requestByUse.VpcId != nil && requestByUse.SubnetId == nil) || (requestByUse.VpcId == nil && requestByUse.SubnetId != nil) {
+			return err
+		}
+		if v, ok := d.GetOk("project_id"); ok {
+			requestByUse.ProjectId = helper.IntInt64(v.(int))
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_instance.go:173]: equestByUse.ProjectId = helper.IntInt64(d.Get("project_id").(int))

project_id有默认值

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 7]
```diff
@@ -51,172 +78,165 @@ func resourceTencentCloudMongodbInstance() *schema.Resource {
...
+		}
+		if v, ok := d.GetOk("security_groups"); ok {
+			securityGroups := v.(*schema.Set).List()
+			requestByUse.SecurityGroup = make([]*string, 0, len(securityGroups))
+			for _, v := range securityGroups {
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_instance.go:178]: requestByUse.SecurityGroup =  helper.InterfacesStringsPoint(v.(*schema.Set).List())

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 8]
```diff
@@ -48,50 +68,88 @@ func (me *MongodbService) ModifyInstanceName(ctx context.Context, instanceId, in
...
 	logId := getLogId(ctx)
-	request := mongodb.NewSetPasswordRequest()
+	request := mongodb.NewResetDBInstancePasswordRequest()
 	request.InstanceId = &instanceId
 	request.UserName = &accountName
 	request.Password = &password
-	ratelimit.Check(request.GetAction())
-	response, err := me.client.UseMongodbClient().SetPassword(request)
+	var response *mongodb.ResetDBInstancePasswordResponse
+	err := resource.Retry(6*writeRetryTimeout, func() *resource.RetryError {
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_mongodb.go:86]: 这个6倍是测出来的吗？

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 9]
```diff
@@ -51,172 +78,165 @@ func resourceTencentCloudMongodbInstance() *schema.Resource {
...
+		ratelimit.Check(request.GetAction())
+		result, e := meta.(*TencentCloudClient).apiV3Conn.UseMongodbClient().CreateDBInstanceHour(request)
+		if e != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, reason:%s\n", logId, request.GetAction(), e.Error())
+			return retryError(e)
+		}
+		response = result
+		return nil
+	})
+	if err != nil {
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_instance.go:206]: var response *mongodb.CreateDBInstanceHourResponse
var err error
	if err = resource.Retry(6*writeRetryTimeout, func() *resource.RetryError {
		ratelimit.Check(request.GetAction())
		if response, err= meta.(*TencentCloudClient).apiV3Conn.UseMongodbClient().CreateDBInstanceHour(request); err != nil {
			log.Printf("[CRITAL]%s api[%s] fail, reason:%s\n", logId, request.GetAction(), e.Error())
			return retryError(err)
		}
		return nil
	}) ; err != nil {
		return err
	}		
	
}

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 10]
```diff
@@ -51,172 +78,165 @@ func resourceTencentCloudMongodbInstance() *schema.Resource {
...
+		return err
 	}
-	if v, ok := d.GetOk("project_id"); ok {
-		projectId := int64(v.(int))
-		request.ProjectId = &projectId
+	log.Printf("[DEBUG]%s api[%s] success, request body [%s], response body [%s]\n",
+		logId, request.GetAction(), request.ToJsonString(), response.ToJsonString())
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_instance.go:210]: 不用打印这个log

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 11]
```diff
@@ -253,30 +301,12 @@ func resourceTencentCloudMongodbInstanceCreate(d *schema.ResourceData, meta inte
...
-		e = fmt.Errorf("mongodb instance status is %d, we won't wait for it finish.", *instance.Status)
-		return resource.NonRetryableError(e)
-	})
-	if err != nil {
-		log.Printf("[CRITAL]%s creating mongodb instance failed, reason:%s\n", logId, err.Error())
-		return err
+	if !has {
+		return fmt.Errorf("[CRITAL]%s creating mongodb instance failed, instance doesn't exist\n", logId)
 	}
 	d.SetId(instanceId)
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_instance.go:311]: 删除 d.SetId(instanceId)

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 12]
```diff
@@ -188,3 +236,89 @@ func (me *MongodbService) DescribeInstancesByFilter(ctx context.Context, instanc
...
+	}
+
+	return
+}
+
+func (me *MongodbService) DescribeAsyncRequestInfo(ctx context.Context, asyncId string) (errRet error) {
+	logId := getLogId(ctx)
+	request := mongodb.NewDescribeAsyncRequestInfoRequest()
+	request.AsyncRequestId = &asyncId
+	err := resource.Retry(10*readRetryTimeout, func() *resource.RetryError {
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_mongodb.go:303]: 时间给的太随意了

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 13]
```diff
@@ -48,50 +68,88 @@ func (me *MongodbService) ModifyInstanceName(ctx context.Context, instanceId, in
 	if err != nil {
 		log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
 			logId, request.GetAction(), request.ToJsonString(), err.Error())
-		return err
+		errRet = err
+		return
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_mongodb.go:72]: 不需要特意拆开，反而累赘

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 14]
```diff
@@ -48,50 +68,88 @@ func (me *MongodbService) ModifyInstanceName(ctx context.Context, instanceId, in
...
+		}
+		response = result
+		return nil
+	})
 	if err != nil {
-		log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
-			logId, request.GetAction(), request.ToJsonString(), err.Error())
-		return err
+		errRet = err
+		return
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_mongodb.go:145]: 同上

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 15]
```diff
@@ -15,12 +18,27 @@ type MongodbService struct {
...
+	err := resource.Retry(20*readRetryTimeout, func() *resource.RetryError {
+		ratelimit.Check(request.GetAction())
+		result, e := me.client.UseMongodbClient().DescribeDBInstances(request)
+		if e != nil {
+			return resource.NonRetryableError(e)
+		}
+
+		if result != nil && result.Response != nil {
+			if len(result.Response.InstanceDetails) != 0 && *result.Response.InstanceDetails[0].Status == MONGODB_INSTANCE_STATUS_PROCESSING {
+				return resource.RetryableError(fmt.Errorf("mongodb instance status is processing"))
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_mongodb.go:35]: DescribeInstanceById  如果实例processing 返回error 不合适   影响read方法

File: tencentcloud/resource_tc_mongodb_sharding_instance.go [hunk 16]
```diff
@@ -397,38 +420,45 @@ func resourceMongodbShardingInstanceUpdate(d *schema.ResourceData, meta interfac
 	d.Partial(true)
 
 	if d.HasChange("memory") || d.HasChange("volume") {
+		// precheck
+		if !(d.HasChange("memory") && d.HasChange("volume")) {
+			return fmt.Errorf("[CRITAL] updating memory and volume of mongodb instance failed, memory and volume must upgrade/downgrade at same time")
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_sharding_instance.go:425]: if !(d.HasChange("memory") && d.HasChange("volume"))这个可以去除下,下面判断oldMemory<newMemory && oldVolume < newVolume就好了

File: tencentcloud/resource_tc_mongodb_sharding_instance.go [hunk 17]
```diff
@@ -397,38 +420,45 @@ func resourceMongodbShardingInstanceUpdate(d *schema.ResourceData, meta interfac
...
-				return nil
+			if !has {
+				return resource.NonRetryableError(fmt.Errorf("[CRITAL]%s updating mongodb sharding instance failed, instance doesn't exist\n", logId))
 			}
-			if *instance.Status == MONGODB_INSTANCE_STATUS_PROCESSING {
-				return resource.RetryableError(fmt.Errorf("mongodb instance status is processing"))
+
+			memoryDes := *infos.Memory / 1024 / (*infos.ReplicationSetNum)
+			volumeDes := *infos.Volume / 1024 / (*infos.ReplicationSetNum)
+			if d.Get("memory").(int) != int(memoryDes) || d.Get("volume").(int) != int(volumeDes) {
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_sharding_instance.go:451]: d.Get("memory").(int)更换成memory

File: tencentcloud/resource_tc_mongodb_sharding_instance.go [hunk 18]
```diff
@@ -397,38 +420,45 @@ func resourceMongodbShardingInstanceUpdate(d *schema.ResourceData, meta interfac
...
 			}
-			e = fmt.Errorf("mongodb instance status is %d, we won't wait for it finish.", *instance.Status)
-			return resource.NonRetryableError(e)
+			return nil
 		})
-		if err != nil {
-			log.Printf("[CRITAL]%s upgrade mongodb instance failed, reason:%s\n", logId, err.Error())
-			return err
+		if errUpdate != nil {
+			return fmt.Errorf("[CRITAL] updating mongodb sharding instance failed, memory and volume values don't change")
```
Comment (Himer) [inline tencentcloud/resource_tc_mongodb_sharding_instance.go:457]: return errUpdate 即可

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 19]
```diff
@@ -188,3 +236,89 @@ func (me *MongodbService) DescribeInstancesByFilter(ctx context.Context, instanc
...
+		if e != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, reason:%s\n", logId, request.GetAction(), e.Error())
+			return retryError(e)
+		}
+		response = result
+		return nil
+	})
+	if err != nil {
+		errRet = err
+		return
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_mongodb.go:262]: return err

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 20]
```diff
@@ -51,172 +78,165 @@ func resourceTencentCloudMongodbInstance() *schema.Resource {
...
 	}
-	if v, ok := d.GetOk("subnet_id"); ok {
-		request.SubnetId = helper.String(v.(string))
+
+	var response *mongodb.CreateDBInstanceHourResponse
+	err := resource.Retry(6*writeRetryTimeout, func() *resource.RetryError {
+		ratelimit.Check(request.GetAction())
+		result, e := meta.(*TencentCloudClient).apiV3Conn.UseMongodbClient().CreateDBInstanceHour(request)
+		if e != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, reason:%s\n", logId, request.GetAction(), e.Error())
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_mongodb_instance.go:200]: log.Printf 不需要自己加 `\n`，他自己会判断如果没有 `\n` 结尾会自己补上
https://github.com/golang/go/blob/master/src/log/log.go#L178

File: tencentcloud/resource_tc_mongodb_standby_instance.go [hunk 21]
```diff
@@ -0,0 +1,565 @@
...
+	}
+
+	instanceId := d.Id()
+
+	_, has, err = mongodbService.DescribeInstanceById(ctx, instanceId)
+	if err != nil {
+		return err
+	}
+	if !has {
+		return fmt.Errorf("[CRITAL]%s creating mongodb standby instance failed, instance doesn't exist\n", logId)
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_mongodb_standby_instance.go:327]: Errorf 不应该以 `\n` 结尾

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 22]
```diff
@@ -5,7 +5,10 @@ import (
 	"fmt"
 	"log"
 
-	mongodb "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/mongodb/v20180408"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+
```
Comment (Sherlock-Holo) [inline tencentcloud/service_tencentcloud_mongodb.go:9]: 没有必要存在的空行

**Comment by brickzzhang** [comment]
Prepaid instances are not allowed to be deleted, so stauts of test cased are `failed`. Plz ignore it.

**Pull request merged**